modules/core/src/main/scala/org/tessellation/Main.scala

```diff
@@ -24,78 +22,90 @@ import org.tessellation.schema.node.NodeState
+                            .asResource
 
-                        _ <- (cli.method match {
-                          case CliMethod.RunValidator =>
-                            storages.node.tryModifyState(NodeState.Initial, NodeState.ReadyToJoin)
-                          case CliMethod.RunGenesis =>
-                            storages.node.tryModifyState(
-                              NodeState.Initial,
-                              NodeState.LoadingGenesis,
-                              NodeState.GenesisReady
-                            ) {
-                              GenesisLoader.make[IO].load(cli.genesisPath).flatMap { accounts =>
-                                logger.info(s"Genesis accounts: ${accounts.show}")
-                              }
-                            } >> services.session.createSession
-                          case _ => IO.raiseError(new RuntimeException("Wrong CLI method"))
-                        }).asResource
-                      } yield ()).useForever
+                          api = HttpApi.make[IO](storages, queues, services, programs, cfg.environment)
+                          _ <- MkHttpServer[IO].newEmber(ServerName("public"), cfg.httpConfig.publicHttp, api.publicApp)
+                          _ <- MkHttpServer[IO].newEmber(ServerName("p2p"), cfg.httpConfig.p2pHttp, api.p2pApp)
+                          _ <- MkHttpServer[IO].newEmber(ServerName("cli"), cfg.httpConfig.cliHttp, api.cliApp)
+
+                          _ <- (method match {
+                            case _: RunValidator =>
+                              storages.node.tryModifyState(NodeState.Initial, NodeState.ReadyToJoin)
+                            case m: RunGenesis =>
+                              storages.node.tryModifyState(
+                                NodeState.Initial,
+                                NodeState.LoadingGenesis,
+                                NodeState.GenesisReady
+                              ) {
+                                GenesisLoader.make[IO].load(m.genesisPath).flatMap { accounts =>
+                                  logger.info(s"Genesis accounts: ${accounts.show}")
+                                }
+                              } >> services.session.createSession
+
+                          }).asResource
+                        } yield ()).useForever
+                      }
                     }
                   }
                 }
               }
             }
-          }
-        }
+        case _ => ExitCode.Error.pure[IO]
+      }
+
     }
 
-  private def loadKeyPair[F[_]: Async: SecurityProvider](cfg: KeyConfig): F[KeyPair] =
+  private def loadKeyPair[F[_]: Async: SecurityProvider](cfg: Run): F[KeyPair] =
     KeyStoreUtils
       .readKeyPairFromStore[F](
-        cfg.keystore,
-        cfg.keyalias.value,
-        cfg.storepass.value.toCharArray,
-        cfg.keypass.value.toCharArray
+        cfg.keyStore.value.toString,
```

Comment:
Is the `.toString` necessary?

Reply:
Yes, it's a Path from fs2 (https://javadoc.io/doc/co.fs2/fs2-io_2.13/latest/fs2/io/file/Path.html).